src/static/modify_assignment.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * Depends on jquery.validate plugin and moment.js
+ * Validates form that edits or add assignments. Then submit it to given endpoint.
+ * Also displays error, spin save icon when loading.
+ * @param formName Name of form to validate
+ * @param postUrl The url to send form data to
+ * @param errorSelector Selector for element to display error
+ * @param saveBtnIcon The selector for icon on save button
+ * @param modalSelector The selector for form modal
+ */
+function modifyAssignment(formName, postUrl, errorSelector, saveBtnIcon, modalSelector) {
+    // custom validators
+    $.validator.addMethod("check_id", function(value, element) {
+        return this.optional(element) || /^[a-zA-Z0-9_\-]+$/.test(value);
+    }, "ID must only contain: a-z A-Z _ -");
```

Comment:
> "ID must only contain: a-z A-Z _ -"

We should include 0-9 in this list right?

Reply:
Nice catch!